File: VB/WindowsApplication3/Custom LayoutControl/MyLayoutControlImplementor.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.Skins
Imports DevExpress.XtraLayout

Namespace DXSample
	Public Class MyLayoutControlImplementor
		Inherits LayoutControlImplementor
		Public Sub New(ByVal layout As ILayoutControlOwner)
			MyBase.New(layout)
		End Sub

		Public Overrides Function CreateLayoutItem(ByVal parent As LayoutGroup) As BaseLayoutItem
			Return New MyLayoutControlItem(CType(parent, LayoutControlGroup))
		End Function

		Public Overrides Function CreateLayoutGroup(ByVal parent As LayoutGroup) As LayoutGroup
			Return New MyLayoutControlGroup(parent)
		End Function
	End Class

End Namespace

File: CS/WindowsApplication3/Custom LayoutControl/MyLayoutControlImplementor.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Skins;
using DevExpress.XtraLayout;

namespace DXSample {
    public class MyLayoutControlImplementor : LayoutControlImplementor
    {
        public MyLayoutControlImplementor(ILayoutControlOwner layout) : base(layout) { }

        public override BaseLayoutItem CreateLayoutItem(LayoutGroup parent)
        {
            return new MyLayoutControlItem((LayoutControlGroup)parent);
        }

        public override LayoutGroup CreateLayoutGroup(LayoutGroup parent)
        {
            return new MyLayoutControlGroup(parent);
        }
    }

}